File: two_sum/two_sum.c
#include <stdio.h>
#include <stdlib.h>

#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))

int *two_sum(int *nums, int length, int target) {
    int *array = (int*)malloc(sizeof(int) * 2);

    for(int index = 0; index < length; index++) {
        for(int j = index + 1; j < length; j++) {
            if(nums[index] + nums[j] == target) {
                array[0] = index;
                array[1] = j;
            }
        }
    }
    
    return array;
}

int main() {
    int array[] = {2,7,11,15};

    int *data = two_sum(array, ARRAY_SIZE(array), 9);

    // output
    // 202 ms	6.2 MB
    printf("%d - %d\n", data[0], data[1]);

    free(data);
    return 0;
}

File: two_sum/two_sum.rb
def two_sum(nums, target)
  hash = {}

  nums.each_with_index do |num, idx|
    result = target - num

    return [idx, hash[result]] if hash.include?(result)

    hash[nums[idx]] = idx
  end
end

# output
# [1, 0]
#
# 90 ms	212 MB
p two_sum([2,7,11,15], 9)
